gengokit/gengokit_test.go

```diff
@@ -0,0 +1,432 @@
+package gengokit
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/TuneLab/go-truss/deftree"
+
+	templateFileAssets "github.com/TuneLab/go-truss/gengokit/template"
+)
+
+func TestTemplatePathToActual(t *testing.T) {
+
+	pathToWants := map[string]string{
+		"NAME-service/":                "package-service/",
+		"NAME-service/test.gotemplate": "package-service/test.go",
+		"NAME-service/NAME-server":     "package-service/package-server",
+	}
+
+	for path, want := range pathToWants {
+		if got := templatePathToActual(path, "package"); got != want {
+			t.Errorf("\n`%v` got\n`%v` wanted", got, want)
+		}
+	}
+}
+
+func TestNewTemplateExecutor(t *testing.T) {
+	const def = `
+		syntax = "proto3";
+
+		// General package
+		package general;
+
+		import "google/api/annotations.proto";
+
+		// RequestMessage is so foo
+		message RequestMessage {
+		  string input = 1;
+		}
+
+		// ResponseMessage is so bar
+		message ResponseMessage {
+		  string output = 1;
+		}
+
+		// ProtoService is a service
+		service ProtoService {
+		  // ProtoMethod is simple. Like a gopher.
+		  rpc ProtoMethod (RequestMessage) returns (ResponseMessage) {
+			// No {} in path and no body, everything is in the query
+			option (google.api.http) = {
+			  get: "/route"
+			};
+		  }
+		}
+	`
+	dt, err := deftree.NewFromString(def)
+	if err != nil {
+		t.Error(err)
+	}
+
+	const goImportPath = "github.com/TuneLab/go-truss/gengokit"
+
+	te, err := newTemplateExecutor(dt, goImportPath)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if got, want := te.ImportPath, goImportPath+"/general-service"; got != want {
+		t.Errorf("\n`%v` was ImportPath\n`%v` was wanted", got, want)
+	}
+
+	if got, want := te.PackageName, dt.GetName(); got != want {
+		t.Errorf("\n`%v` was PackageName\n`%v` was wanted", got, want)
+	}
+}
+
+func TestGetProtoService(t *testing.T) {
+	const def = `
+		syntax = "proto3";
+
+		// General package
+		package general;
+
+		import "google/api/annotations.proto";
+
+		// RequestMessage is so foo
+		message RequestMessage {
+		  string input = 1;
+		}
+
+		// ResponseMessage is so bar
+		message ResponseMessage {
+		  string output = 1;
+		}
+
+		// ProtoService is a service
+		service ProtoService {
+		  // ProtoMethod is simple. Like a gopher.
+		  rpc ProtoMethod (RequestMessage) returns (ResponseMessage) {
+			// No {} in path and no body, everything is in the query
+			option (google.api.http) = {
+			  get: "/route"
+			};
+		  }
+		}
+	`
+	dt, err := deftree.NewFromString(def)
+	if err != nil {
+		t.Error(err)
```

Comment:
`testing.T.Error` will fail the test but not stop test execution. If you want to stop test execution, use `testing.T.Fail`.